File: frontend/src/api/article.ts
import { client, type ApiResponse } from "@/api/client";

export interface ArticleItem {
  id: number;
  title: string;
  summary: string;
  type: number;
  cover: string;
  cover_type: string;
  menu_id: number;
  channel_id: number;
  sort_order: number;
  is_paid: number;
  admin_id: number;
  is_top: number;
  is_hot: number;
  is_recommend: number;
  comment_num: number;
  share_num: number;
  view_num: number;
  collect_num: number;
  status: number;
  created_at: string;
  content?: { content: string };
}

export interface ArticleListResult {
  list: ArticleItem[];
  total: number;
  page: number;
  page_size: number;
}

export interface ArticleBatchCreateItem {
  title: string;
  summary: string;
  cover?: string;
  cover_type?: string;
  content: string;
}

export interface ArticleBatchCreatePayload {
  type: number;
  cover: string;
  cover_type: string;
  menu_id: number;
  channel_id: number;
  sort_order?: number;
  is_paid?: number;
  is_top?: number;
  is_hot?: number;
  is_recommend?: number;
  status: number;
  items: ArticleBatchCreateItem[];
}

export const articleListAPI = (params: Record<string, unknown>) => client.get<any, ApiResponse<ArticleListResult>>("/articles", { params });
export const articleDetailAPI = (id: string | number) => client.get<any, ApiResponse<ArticleItem>>(`/articles/${id}`);
export const articleCreateAPI = (data: Record<string, unknown>) => client.post<any, ApiResponse<{ id: number }>>("/articles", data);
export const articleBatchCreateAPI = (data: ArticleBatchCreatePayload) => client.post<any, ApiResponse<{ created: number; ids: number[] }>>("/articles/batch", data);
export const articleUpdateAPI = (id: string | number, data: Record<string, unknown>) => client.put<any, ApiResponse<{ id: number }>>(`/articles/${id}`, data);
export const articleStatusAPI = (data: { ids: number[]; status: number }) => client.put<any, ApiResponse<{ updated: number }>>("/articles/status", data);
export const articleDeleteAPI = (ids: number[]) => client.delete<any, ApiResponse<{ deleted: number }>>("/articles", { data: { ids } });

File: backend/internal/dto/article.go
package dto

type ArticleQuery struct {
	PageQuery
	Title       string `form:"title"`
	Type        *int   `form:"type"`
	MenuID      *int64 `form:"menu_id"`
	ChannelID   *int64 `form:"channel_id"`
	IsPaid      *int   `form:"is_paid"`
	Status      *int   `form:"status"`
	CreatedFrom string `form:"created_from"`
	CreatedTo   string `form:"created_to"`
}

type ArticleUpsertRequest struct {
	Title       string  `json:"title" binding:"required"`
	Summary     string  `json:"summary"`
	Type        int     `json:"type" binding:"required"`
	Cover       string  `json:"cover"`
	CoverLarge  string  `json:"cover_large"`
	CoverMedium string  `json:"cover_medium"`
	CoverSmall  string  `json:"cover_small"`
	CoverType   string  `json:"cover_type"`
	MenuID      int64   `json:"menu_id"`
	MenuIDs     []int64 `json:"menu_ids"`
	ChannelID   int64   `json:"channel_id" binding:"required"`
	SortOrder   int8    `json:"sort_order"`
	IsPaid      int     `json:"is_paid"`
	IsTop       int     `json:"is_top"`
	IsHot       int     `json:"is_hot"`
	IsRecommend int     `json:"is_recommend"`
	Status      int     `json:"status"`
	Content     string  `json:"content" binding:"required"`
}

type ArticleBatchCreateItem struct {
	Title       string `json:"title" binding:"required"`
	Summary     string `json:"summary"`
	Cover       string `json:"cover"`
	CoverLarge  string `json:"cover_large"`
	CoverMedium string `json:"cover_medium"`
	CoverSmall  string `json:"cover_small"`
	CoverType   string `json:"cover_type"`
	Content     string `json:"content" binding:"required"`
}

type ArticleBatchCreateRequest struct {
	Type        int                      `json:"type" binding:"required"`
	Cover       string                   `json:"cover"`
	CoverLarge  string                   `json:"cover_large"`
	CoverMedium string                   `json:"cover_medium"`
	CoverSmall  string                   `json:"cover_small"`
	CoverType   string                   `json:"cover_type"`
	MenuID      int64                    `json:"menu_id" binding:"required"`
	MenuIDs     []int64                  `json:"menu_ids"`
	ChannelID   int64                    `json:"channel_id" binding:"required"`
	SortOrder   int8                     `json:"sort_order"`
	IsPaid      int                      `json:"is_paid"`
	IsTop       int                      `json:"is_top"`
	IsHot       int                      `json:"is_hot"`
	IsRecommend int                      `json:"is_recommend"`
	Status      int                      `json:"status"`
	Items       []ArticleBatchCreateItem `json:"items" binding:"required,min=1"`
}

type ArticleBatchStatusRequest struct {
	IDs    []int64 `json:"ids" binding:"required,min=1"`
	Status int     `json:"status" binding:"required"`
}
